### PR TITLE
[expo-constants][js] Ensure classic manifest originalFullName is used over id

### PR DESCRIPTION
**Pull request by wschurman**
# Why

This PR audits all `Constants.manifest` callsites and ensures that `originalFullName` is used instead of `id` when possible.

# How

Add fields to manifest type for docblocks. Grep for `Constants.manifest` (and variants).

# Test Plan

`expo cp`

### CODE REVIEW

File: packages/expo-constants/src/Constants.types.ts [hunk 1]
```diff
@@ -78,6 +78,39 @@ export interface AppManifest extends ExpoConfig {
   debuggerHost?: string;
   mainModuleName?: string;
   logUrl?: string;
+
+  /**
+   * The Expo account name and slug for this project.
+   * @deprecated - Prefer `projectId` or `originalFullName` instead for scoping due to immutability.
```
Comment (ide) [inline packages/expo-constants/src/Constants.types.ts:84]:
```suggestion
   * @deprecated - Prefer `projectId` or `originalFullName` instead for identification and `scopeKey` for
   *   scoping due to immutability.
```

**Review by ide — COMMENTED** [review]
Since we maintain NCL and Home, it'd be nice to read only originalFullName and not fall back to id. However, for this to happen, Expo CLI would need to send originalFullName in its development manifests and I"m not sure it does that.

**Comment by wschurman** [comment]
> Since we maintain NCL and Home, it'd be nice to read only originalFullName and not fall back to id. However, for this to happen, Expo CLI would need to send originalFullName in its development manifests and I"m not sure it does that.

As of https://github.com/expo/expo-cli/issues/3494 I'm pretty sure `originalFullName` is in the manifests. I'll investigate removing `id` from NCL and Home in a follow-up PR since I'm not 100% sure it is possible in all cases.

**Review by ide — APPROVED** [review]
Consider adding comments noting our intent to remove the Constants.manifest.id accesses.

**Comment by wschurman** [comment]
> Consider adding comments noting our intent to remove the Constants.manifest.id accesses.

https://linear.app/expo/issue/ENG-1150/investigate-removing-id-from-home-and-ncl

**Pull request merged**